kostyor/upgrades/engine.py

```diff
@@ -152,17 +152,21 @@ def start(self):
         # keystone, then with nova, and so on. See iteration details in
         # get_controllers() docstring.
         for host in get_controllers(hosts):
+            self.driver.pre_host_upgrade_hook(host)
```

Comment:
Maybe submit returned value to tasks? So driver developer can return a certain Celery task/chain/group to do some long-running stuff?